### PR TITLE
Added the ctrl_state param to the mcp() function & MCPhaseGate() class

### PR DESCRIPTION
**Pull request by elucidator8918**
### Description:

Currently, the mcp function lacks a ctrl_state parameter, limiting its flexibility in certain quantum operations. 

The proposed change modifies the mcp function to include a ctrl_state parameter, allowing users to specify the control state of the multi-controlled Phase gate. This parameter can accept either a decimal value or a bitstring and defaults to controlling the '1' state if not provided.

### Implementation Details:
- Modified the mcp function to accept the ctrl_state parameter.
- Added Release notes.
- Added tests for checking the consistency of mcp() function.

@ShellyGarion: Please review the changes. Thank you :) Again!

### CODE REVIEW

File: releasenotes/notes/add-ctrl_state-mcp-parameter-b23562aa7047665a.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+---
+features:
+  - |
+    Added `ctrl_state` parameter to :func:`QuantumCircuit.mcp()`.
```
Comment (1ucian0) [inline releasenotes/notes/add-ctrl_state-mcp-parameter-b23562aa7047665a.yaml]:
```suggestion
    Added ``ctrl_state`` parameter to :func:`QuantumCircuit.mcp()`.
```

File: releasenotes/notes/add-ctrl_state-mcp-parameter-b23562aa7047665a.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+
+    The :func:`QuantumCircuit.mcp()` function in the quantum circuit library has
+    been enhanced to include a `ctrl_state` parameter, allowing users to specify
```
Comment (1ucian0) [inline releasenotes/notes/add-ctrl_state-mcp-parameter-b23562aa7047665a.yaml]:
```suggestion
    been enhanced to include a ``ctrl_state`` parameter, allowing users to specify
```

**Comment by ShellyGarion** [comment]
Note that this PR does not only add the `ctrl_state` param to the `mcp()` function, but it also adds it to the `MCPhaseGate()`.
This should be emphasized in the release notes.
In addition, would you be able to add some test cases to check that it generates a valid gate for a not-all-ones control state?

**Comment by ShellyGarion** [comment]
The test that you added is good, but it only compares the results of `MCPhaseGate()` and `mcp()`. 
Could you add some test that `MCPhaseGate()` with open controls behave as expected? 
In the test file, there are many tests for open controls for `MCXGate()`, could you add similar tests for `MCPhaeGate()` ?
See for example:
https://github.com/Qiskit/qiskit/blob/5fa47269f022abd031ad178eb77359eb9544d6fe/test/python/circuit/test_controlled_gate.py#L1108

**Comment by elucidator8918** [comment]
@ShellyGarion added the test!

**Review by ShellyGarion — APPROVED** [review]
LGTM.
@elucidator8918 - thanks for your contribution to qiskit, nice work!

**Review by alexanderivrii — APPROVED** [review]
LGTM!

**Pull request merged**